Updated Resource Pack Formats

Mosnar23 opened the pull request.

Mosnar23 committed:
Updated Resource Pack Formats

Trial97 commented:
Can you please follow the instructions from here https://github.com/PrismLauncher/PrismLauncher/pull/4242/checks?check_run_id=53234147903 to sign your commit?

Mosnar23 commented:
Should I close and re-open the pull request? (And sign my commit this time?)

Mosnar23 commented:
From the Linux AppImage from GitHub Actions, it looks like my change works!
(See #4237 for before image)
<img width="1034" height="599" alt="image" src="https://github.com/user-attachments/assets/c229cbe1-39cf-4bd1-9fb1-5561a4ff5f8c" />

Mosnar23 committed:
Updated Data Pack Formats so that I can sign this commit. Speaking of which,
DCO Remediation Commit for Mosnar23 <mosnar23@icloud.com>

I, Mosnar23 <mosnar23@icloud.com>, hereby add my Signed-off-by to this commit: 67d9cc961f559fb3414b4260aa6ac576c94a9eb4

Signed-off-by: Mosnar23 <mosnar23@icloud.com>

TheKodeToad commented on launcher/minecraft/mod/ResourcePack.cpp:
```diff
@@ -24,7 +24,29 @@ static const QMap<int, std::pair<Version, Version>> s_pack_format_versions = {
+    { 62, { Version("25w21a"), Version("25w21a") } },      { 63, { Version("1.21.6-pre1"), Version("1.21.7-rc1") } },
+    { 64, { Version("1.21.7-rc2"), Version("1.21.8") } },  { 65.0, { Version("25w31a"), Version("25w31a") } },
+    { 65, { Version("25w31a"), Version("25w31a") } },      { 65.1, { Version("25w32a"), Version("25w32a") } },
+    { 65.2, { Version("25w33a"), Version("25w33a") } },    { 66.0, { Version("25w34a"), Version("25w34b") } },
```
This won't work properly as this map has int keys

Mosnar23 replied:
Ok, I see the problem. Thanks for bringing this up. Should I just remove formats with decimals in their numbers?

TheKodeToad replied:
Minecraft seems to have both Major and Minor pack versions. The correct fix would be to detect that, but you'd need to modify pack.mcmeta parsing.. I'm not even sure how it's formatted in that file

Mosnar23 commented:
Since I've made this pull request, 2 new pack formats have been created, 70.1 and 71. I think the best solution would be to have the launcher pull pack formats live from [the official list of resource pack formats](https://minecraft.wiki/w/Pack_format#List_of_resource_pack_formats), but even that seems to be a bit out of date. Also, I don't think that I could do that because I have no idea how c++ works, I was just doing this because I saw a seemingly simple and also out of date list.

Trial97 commented:
closed in favor of https://github.com/PrismLauncher/PrismLauncher/pull/5199
